CCP-684: added support for ctaPrimary and ctaPrimaryContrast to AddTo…

aaron-martin opened the pull request:
@sznowicki 
Please let me know if you are not able to do this review.

Thanks
Aaron

sznowicki commented:
Almost. This variables should be always available, therefore there is no need to check if it should be used.

The variables should be prepared in `THEME_ROOT/theme-config.js`. There should be the logic that makes the merge of incoming ctaPrimary/ctaPrimaryContrast from the shop settings or assignes the primary/...Contrast to ctaPrimary. 

In pseudocode  it would be like this:

```js
// Assign cta and ctaContrast hex from primary and primaryContrast which comes from shop/extension config
let cta = themeConfig.primary;
let ctaContrast = themeConfig.primaryContrast

// If extension config has both of them configures, assign them instead
if (themeConfig.ctaPrimary && themeConfig.ctaPrimaryContrast) {
   cta = themeConfig.cta
  ctaContrast = themeConfig.ctaContrast
}

```

Additionally, after some thinking, I think we should maybe call them `cta` and `ctaContrast`. I think we fixed on the primary since it falls back to it, but the `ctaPrimary` name is maybe not the best idea.

sznowicki commented:
Long tory short (I have tendency to put too much information in my comments):

1. What you did is almost fine, but please use less logic, simply use `cta` color variable, no `primary` any more there
2. `colors.cta` and `colors.ctaContrast` is *always* available in themes since it is prepared in theme-config.js (or however it's called, I cannot really check it now)

aaron-martin commented:
@sznowicki 
I changed the name of the new colors properties to cta and ctaContrast. I removed logic to use primary colors when cta and ctaContrast are not available.

sznowicki commented:
Theme snapshots needs to be updated with library changes. I'll do that.

richardgorman commented on libraries/ui-shared/Button/index.jsx:
```diff
@@ -8,6 +8,7 @@ const buttonTypes = [
   'regular',
   'primary',
   'secondary',
+  'checkout',
```
Generic components should not have contextual options inside.

sznowicki replied:
What would you suggest instead for this case?
